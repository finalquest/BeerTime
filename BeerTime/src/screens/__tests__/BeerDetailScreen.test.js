import React from 'react';
import { shallow } from 'enzyme';

import Screen from '../BeerDetailScreen';

describe('BeerDetailScreen test suite', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<Screen />);
    expect(wrapper).toMatchSnapshot();
  });
});
