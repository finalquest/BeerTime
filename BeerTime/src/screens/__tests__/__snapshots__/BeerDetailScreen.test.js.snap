// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerDetailScreen test suite renders as expected 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <BackArrow
    styleImage={
      Object {
        "alignSelf": "center",
        "tintColor": "black",
      }
    }
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "marginHorizontal": 10,
      }
    }
  >
    <Label
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 24,
          "fontWeight": "500",
        }
      }
      uppercase={false}
    >
      name
    </Label>
    <Label
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 14,
        }
      }
      uppercase={false}
    >
      tagline
    </Label>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#F2FCF7",
          "borderColor": "#02C66A",
          "borderRadius": 5,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginVertical": 20,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 15,
        }
      }
    >
      <Label
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "fontWeight": "400",
              "marginRight": 3,
              "textAlign": "left",
            },
            Object {
              "marginVertical": 5,
            },
          ]
        }
        uppercase={false}
      >
        description
      </Label>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#F2FCF7",
          "borderColor": "#02C66A",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 15,
        }
      }
    >
      <PropertyLabel
        Component={[Function]}
        ellipsizeMode="tail"
        fontSize={14}
        name="ABV"
        nameStyles={
          Array [
            Object {
              "alignSelf": "stretch",
              "fontWeight": "400",
              "marginRight": 3,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "600",
            },
          ]
        }
        numberOfLines={0}
        separator=":"
        style={
          Object {
            "marginVertical": 5,
          }
        }
        value="abv"
        valueStyles={
          Object {
            "alignSelf": "stretch",
            "fontWeight": "400",
            "marginRight": 3,
            "textAlign": "left",
          }
        }
      />
      <PropertyLabel
        Component={[Function]}
        ellipsizeMode="tail"
        fontSize={14}
        name="Volume"
        nameStyles={
          Array [
            Object {
              "alignSelf": "stretch",
              "fontWeight": "400",
              "marginRight": 3,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "600",
            },
          ]
        }
        numberOfLines={0}
        separator=":"
        style={
          Object {
            "marginVertical": 5,
          }
        }
        value="20 unit"
        valueStyles={
          Object {
            "alignSelf": "stretch",
            "fontWeight": "400",
            "marginRight": 3,
            "textAlign": "left",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#F2FCF7",
          "borderColor": "#02C66A",
          "borderRadius": 5,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginVertical": 20,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 15,
        }
      }
    >
      <PropertyLabel
        Component={[Function]}
        ellipsizeMode="tail"
        fontSize={14}
        name="Food Pairing"
        nameStyles={
          Array [
            Object {
              "alignSelf": "stretch",
              "fontWeight": "400",
              "marginRight": 3,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "600",
            },
          ]
        }
        numberOfLines={0}
        separator=":"
        style={
          Object {
            "flex": 1,
            "marginVertical": 5,
          }
        }
        value="food"
        valueStyles={
          Array [
            Object {
              "alignSelf": "stretch",
              "fontWeight": "400",
              "marginRight": 3,
              "textAlign": "left",
            },
            Object {
              "flex": 1,
            },
          ]
        }
      />
    </View>
  </ScrollViewMock>
</View>
`;

exports[`BeerDetailScreen test suite renders as expected without food_pairing 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <BackArrow
    styleImage={
      Object {
        "alignSelf": "center",
        "tintColor": "black",
      }
    }
  />
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "marginHorizontal": 10,
      }
    }
  >
    <Label
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 24,
          "fontWeight": "500",
        }
      }
      uppercase={false}
    >
      name
    </Label>
    <Label
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 14,
        }
      }
      uppercase={false}
    >
      tagline
    </Label>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#F2FCF7",
          "borderColor": "#02C66A",
          "borderRadius": 5,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginVertical": 20,
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 15,
        }
      }
    >
      <Label
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignSelf": "stretch",
              "fontWeight": "400",
              "marginRight": 3,
              "textAlign": "left",
            },
            Object {
              "marginVertical": 5,
            },
          ]
        }
        uppercase={false}
      >
        description
      </Label>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#F2FCF7",
          "borderColor": "#02C66A",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingHorizontal": 10,
          "paddingVertical": 15,
        }
      }
    >
      <PropertyLabel
        Component={[Function]}
        ellipsizeMode="tail"
        fontSize={14}
        name="ABV"
        nameStyles={
          Array [
            Object {
              "alignSelf": "stretch",
              "fontWeight": "400",
              "marginRight": 3,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "600",
            },
          ]
        }
        numberOfLines={0}
        separator=":"
        style={
          Object {
            "marginVertical": 5,
          }
        }
        value="abv"
        valueStyles={
          Object {
            "alignSelf": "stretch",
            "fontWeight": "400",
            "marginRight": 3,
            "textAlign": "left",
          }
        }
      />
      <PropertyLabel
        Component={[Function]}
        ellipsizeMode="tail"
        fontSize={14}
        name="Volume"
        nameStyles={
          Array [
            Object {
              "alignSelf": "stretch",
              "fontWeight": "400",
              "marginRight": 3,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "600",
            },
          ]
        }
        numberOfLines={0}
        separator=":"
        style={
          Object {
            "marginVertical": 5,
          }
        }
        value="20 unit"
        valueStyles={
          Object {
            "alignSelf": "stretch",
            "fontWeight": "400",
            "marginRight": 3,
            "textAlign": "left",
          }
        }
      />
    </View>
  </ScrollViewMock>
</View>
`;
